staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -47,7 +47,7 @@ func WithAuthentication(handler http.Handler, auth authenticator.Request, failed
 			req = req.WithContext(authenticator.WithAudiences(req.Context(), apiAuds))
 		}
 		resp, ok, err := auth.AuthenticateRequest(req)
-		defer recordAuthMetrics(req.Context(), resp, ok, err, apiAuds, authenticationStart)
+		defer recordAuthMetrics(req.Context(), resp, ok, err, apiAuds, authenticationStart, time.Now())
```

Comment:
Why do we need to defer here?

Reply:
I don't think we need it. I could remove defer too.

Reply:
Done.